File: scripts/diag6.py
#!/usr/bin/env python3
"""Check actual DB columns to find column mismatches in emailProcessor."""
import re, subprocess, json

with open('/Users/user/leadflow/.env') as f:
    text = f.read()

anon = re.search(r'NEXT_PUBLIC_SUPABASE_ANON_KEY="([^"]+)"', text).group(1)
srk = re.search(r'SUPABASE_SERVICE_ROLE_KEY="([^"]+)"', text).group(1)
url = 'https://eqksgmbcyvfllcaeqgbj.supabase.co/rest/v1'

cid = 'd989f357-39cb-4d99-a3cd-56a36e2562be'
oid = '64209895-565d-4974-9d41-3f39d1a1b467'

# 1) Get actual campaigns columns via select ALL
print("1) CAMPAIGNS table - actual columns:")
r = subprocess.run(['curl', '-s',
    f'{url}/campaigns?id=eq.{cid}&limit=1',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, list) and len(data) > 0:
        cols = list(data[0].keys())
        print(f"   Column count: {len(cols)}")
        print(f"   Columns: {cols}")
        # Check specific columns used in emailProcessor
        check = ['steps', 'use_powersend', 'powersend_config', 'powersend_server_ids', 
                 'sender_id', 'sender_ids', 'config', 'sent_count', 'total_leads', 'org_id']
        for col in check:
            exists = col in cols
            val = data[0].get(col, 'N/A')
            if col == 'steps':
                val = f"{len(val) if isinstance(val, list) else 'N/A'} steps"
            elif isinstance(val, (dict, list)):
                val = json.dumps(val)[:100]
            print(f"   {'✓' if exists else '✗'} {col}: {val}")
except Exception as e:
    print(f"   Error: {e}\n   {r.stdout[:300]}")

# 2) Get actual leads columns
print("\n2) LEADS table - actual columns (sample):")
# Get the test lead
r = subprocess.run(['curl', '-s',
    f'{url}/campaign_recipients?campaign_id=eq.{cid}&status=eq.active&select=lead_id&limit=1',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
lead_id = json.loads(r.stdout)[0]['lead_id']
r = subprocess.run(['curl', '-s',
    f'{url}/leads?id=eq.{lead_id}&limit=1',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, list) and len(data) > 0:
        cols = list(data[0].keys())
        print(f"   Column count: {len(cols)}")
        print(f"   Columns: {cols}")
        # Check columns used in emailProcessor select
        check = ['email', 'first_name', 'last_name', 'company', 'job_title', 'city', 'country',
                 'timezone', 'phone', 'custom_fields', 'tags', 'status', 'org_id', 'source', 'sentiment']
        for col in check:
            exists = col in cols
            print(f"   {'✓' if exists else '✗'} {col}")
except Exception as e:
    print(f"   Error: {e}\n   {r.stdout[:300]}")

# 3) Get actual organizations columns 
print("\n3) ORGANIZATIONS table - actual columns:")
r = subprocess.run(['curl', '-s',
    f'{url}/organizations?id=eq.{oid}&limit=1',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, list) and len(data) > 0:
        cols = list(data[0].keys())
        print(f"   Column count: {len(cols)}")
        print(f"   Columns: {cols}")
        check = ['name', 'ai_usage_current', 'subscription_status', 'plan_tier', 'smart_sending_enabled']
        for col in check:
            exists = col in cols
            print(f"   {'✓' if exists else '✗'} {col}")
except Exception as e:
    print(f"   Error: {e}\n   {r.stdout[:300]}")

# 4) Check the test event we sent - did it change the recipient?
print(f"\n4) Test recipient (lead {lead_id}):")
r = subprocess.run(['curl', '-s',
    f'{url}/campaign_recipients?campaign_id=eq.{cid}&lead_id=eq.{lead_id}',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, list) and len(data) > 0:
        print(f"   {json.dumps(data[0], indent=2)}")
except:
    print(f"   {r.stdout[:300]}")

# 5) Test individual queries that emailProcessor does
print("\n5) Testing emailProcessor queries individually:")

# 5a) Campaign select (same columns as emailProcessor)
print("  5a) Campaign fetch:")
r = subprocess.run(['curl', '-s',
    f'{url}/campaigns?id=eq.{cid}&select=id,steps,status,use_powersend,powersend_config,powersend_server_ids,sender_id,sender_ids,config,name,org_id,sent_count,total_leads',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, dict) and 'code' in data:
        print(f"   ✗ ERROR: {data.get('message')} | hint: {data.get('hint')}")
    else:
        print(f"   ✓ OK ({len(data)} rows)")
except:
    print(f"   {r.stdout[:200]}")

# 5b) Lead fetch
print("  5b) Lead fetch:")
r = subprocess.run(['curl', '-s',
    f'{url}/leads?id=eq.{lead_id}&select=id,email,first_name,last_name,company,job_title,city,country,timezone,phone,custom_fields,tags,status,org_id,source,sentiment',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, dict) and 'code' in data:
        print(f"   ✗ ERROR: {data.get('message')} | hint: {data.get('hint')}")
    else:
        print(f"   ✓ OK ({len(data)} rows)")
except:
    print(f"   {r.stdout[:200]}")

# 5c) Organization fetch
print("  5c) Organization fetch:")
r = subprocess.run(['curl', '-s',
    f'{url}/organizations?id=eq.{oid}&select=id,name,ai_usage_current,subscription_status,plan_tier,smart_sending_enabled',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, dict) and 'code' in data:
        print(f"   ✗ ERROR: {data.get('message')} | hint: {data.get('hint')}")
    else:
        print(f"   ✓ OK ({len(data)} rows)")
except:
    print(f"   {r.stdout[:200]}")

# 5d) Recipient fetch
print("  5d) Recipient fetch:")
r = subprocess.run(['curl', '-s',
    f'{url}/campaign_recipients?campaign_id=eq.{cid}&lead_id=eq.{lead_id}&select=id,status,current_step,last_sent_at',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, dict) and 'code' in data:
        print(f"   ✗ ERROR: {data.get('message')} | hint: {data.get('hint')}")
    else:
        print(f"   ✓ OK ({len(data)} rows)")
except:
    print(f"   {r.stdout[:200]}")

# 5e) Analytics (subscription check)
print("  5e) Analytics daily fetch:")
r = subprocess.run(['curl', '-s',
    f'{url}/analytics_daily?org_id=eq.{oid}&date=gte.2026-04-01&select=sent_count',
    '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}'
], capture_output=True, text=True)
try:
    data = json.loads(r.stdout)
    if isinstance(data, dict) and 'code' in data:
        print(f"   ✗ ERROR: {data.get('message')} | hint: {data.get('hint')}")
    else:
        print(f"   ✓ OK ({len(data)} rows)")
except:
    print(f"   {r.stdout[:200]}")

# 5f) Check RPCs
print("  5f) RPCs:")
for rpc_name, params in [
    ('increment_campaign_stat', {'campaign_id_param': cid, 'column_param': 'sent_count'}),
    ('increment_server_usage', {'server_id_param': '7a0741c6-4678-4863-9a1e-26c888e5fae4'}),
    ('increment_mailbox_usage', {'mailbox_id_param': '2320ab0b-2a0c-4e23-9c37-870ab558f96d'}),
]:
    r = subprocess.run(['curl', '-s', '-o', '/dev/null', '-w', '%{http_code}',
        '-X', 'POST',
        f'https://eqksgmbcyvfllcaeqgbj.supabase.co/rest/v1/rpc/{rpc_name}',
        '-H', f'apikey: {anon}', '-H', f'Authorization: Bearer {srk}',
        '-H', 'Content-Type: application/json',
        '-d', json.dumps(params)
    ], capture_output=True, text=True)
    status = r.stdout.strip()
    print(f"   {rpc_name}: HTTP {status}")

File: scripts/diag6.mjs
import { createClient } from '@supabase/supabase-js';

const sb = createClient(
  'https://eqksgmbcyvfllcaeqgbj.supabase.co',
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImVxa3NnbWJjeXZmbGxjYWVxZ2JqIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2OTg1MTQ2MCwiZXhwIjoyMDg1NDI3NDYwfQ.fwK_E8MU6JkVaJJRYN3WBu0dAQTaXKr2r76MnAxr_aI'
);

const CID = '9d7ae459-72cf-4a5c-8390-4a1242ff3405';
const ORG = '64209895-565d-4974-9d41-3f39d1a1b467';

async function main() {
  // 1. Check for ANY activity_log entries for this campaign recently
  console.log('=== Recent activity_log for this campaign ===');
  const { data: activities } = await sb.from('activity_log')
    .select('action_type, description, created_at')
    .eq('org_id', ORG)
    .order('created_at', { ascending: false })
    .limit(20);
  for (const a of activities || []) {
    console.log(`  [${a.created_at}] ${a.action_type}: ${a.description?.slice(0, 100)}`);
  }

  // 2. Check subscription state  
  console.log('\n=== Subscription check ===');
  const { data: org } = await sb.from('organizations')
    .select('subscription_status, plan_tier, ai_usage_current')
    .eq('id', ORG).single();
  console.log(JSON.stringify(org));

  // 3. Monthly email volume vs limits
  const firstOfMonth = new Date();
  firstOfMonth.setDate(1);
  const { data: usage } = await sb.from('analytics_daily')
    .select('sent_count, date')
    .eq('org_id', ORG)
    .gte('date', firstOfMonth.toISOString().split('T')[0]);
  const monthlyVolume = (usage || []).reduce((acc, curr) => acc + (curr.sent_count || 0), 0);
  console.log(`Monthly volume: ${monthlyVolume}`);
  console.log('Daily usage:', JSON.stringify(usage));

  // 4. Test the RPCs that emailProcessor uses
  console.log('\n=== Test get_next_powersend_node ===');
  const serverIds = ['1dac6ece-a3be-401a-840a-34072d90be2c', '7a0741c6-4678-4863-9a1e-26c888e5fae4'];
  const { data: node, error: nodeErr } = await sb
    .rpc('get_next_powersend_node', { org_id_param: ORG, server_ids_param: serverIds })
    .single();
  if (nodeErr) console.log('ERROR:', nodeErr.message);
  else console.log(`Node: ${node?.name} (${node?.id?.slice(0,8)}) usage: ${node?.current_usage}/${node?.daily_limit}`);

  if (node) {
    console.log('\n=== Test get_next_pool_mailbox ===');
    const { data: mb, error: mbErr } = await sb
      .rpc('get_next_pool_mailbox', { server_id_param: node.id })
      .single();
    if (mbErr) console.log('ERROR:', mbErr.message);
    else console.log(`Mailbox: ${mb?.email} (smtp: ${mb?.smtp_host}:${mb?.smtp_port})`);

    // 5. Try a real SMTP connection test
    if (mb) {
      console.log('\n=== SMTP connection test ===');
      const nodemailer = await import('nodemailer');
      try {
        const transport = nodemailer.default.createTransport({
          host: mb.smtp_host,
          port: mb.smtp_port || 465,
          secure: (mb.smtp_port || 465) === 465,
          auth: { user: mb.smtp_username, pass: mb.smtp_password },
          connectionTimeout: 10000,
          socketTimeout: 10000,
        });
        const verified = await transport.verify();
        console.log('SMTP verify result:', verified);
        transport.close();
      } catch (err) {
        console.log('SMTP connection FAILED:', err.message);
      }
    }
  }

  // 6. Check increment_campaign_stat RPC exists
  console.log('\n=== Test increment_campaign_stat ===');
  try {
    // Just test the RPC exists (will increment by 0 effectively)
    const { error } = await sb.rpc('increment_campaign_stat', {
      campaign_id_param: CID,
      column_param: 'sent_count'
    });
    if (error) console.log('RPC ERROR:', error.message);
    else console.log('RPC works (note: this incremented sent_count by 1, may need to decrement)');
  } catch (err) {
    console.log('RPC call failed:', err.message);
  }

  // 7. Check for leads with issues for this campaign
  console.log('\n=== Sample lead check ===');
  const { data: sampleRecip } = await sb.from('campaign_recipients')
    .select('lead_id')
    .eq('campaign_id', CID)
    .eq('status', 'active')
    .is('last_sent_at', null)
    .limit(1);
  if (sampleRecip?.[0]) {
    const { data: lead } = await sb.from('leads')
      .select('id, email, first_name, company, org_id')
      .eq('id', sampleRecip[0].lead_id)
      .single();
    console.log('Sample lead:', JSON.stringify(lead));
    if (!lead) console.log('>>> LEAD NOT FOUND — leads may have been deleted!');
  }
}

main().catch(console.error);
